app/controllers/admin/categories_controller.rb

```diff
@@ -2,6 +2,9 @@ class Admin::CategoriesController < ApplicationController
 
   before_filter :ensure_is_admin
 
+  #Allow admin to access admin panel before email confirmation
+  skip_filter :cannot_access_without_confirmation
```

Comment:
Understood. Ok to take it one step at a time. Good point about AdminController and UserController names having a possibility for confusion. How about AdminPageController ?